src/lib/Farm.js

```diff
@@ -281,34 +302,70 @@ class AutomationFarm
                 continue;
             }
 
+            // Cant harvest berries if the plant is not fully ripe
             if (plot.stage() != PlotStage.Berry)
             {
                 continue;
             }
 
-            // Harvest berry in any of those cases:
-            //   - No strategy is currently active
-            //   - The unlock feature is disabled
-            //   - Another feature required force harvesting
-            //   - The strategy requires to harvest as soon as possible
-            //   - The berry is the target one
-            //   - The berry is close to dying (less than 15s)
-            if ((this.__internal__currentStrategy === null)
-                || ((this.__internal__currentStrategy.harvestStrategy !== this.__internal__harvestTimingType.LetTheBerryDie)
-                    && ((Automation.Utils.LocalStorage.getValue(this.Settings.FocusOnUnlocks) === "false")
-                        || this.ForcePlantBerriesAsked
-                        || (this.__internal__currentStrategy.harvestStrategy === this.__internal__harvestTimingType.AsSoonAsPossible)
-                        || ((this.__internal__currentStrategy.berryToUnlock !== undefined)
-                            && (this.__internal__currentStrategy.berryToUnlock == plot.berry))
-                        || ((plot.berryData.growthTime[PlotStage.Berry] - plot.age) < 15))))
+            // Handle harvesting according to the current strategy
+            if (this.__internal__currentStrategy !== null && focusOnUnlocks)
             {
+                // If the strategy requires to let the berries die, never harvest them
+                if (this.__internal__currentStrategy.harvestStrategy === this.__internal__harvestTimingType.LetTheBerryDie)
+                {
+                    continue;
+                }
+                // If the strategy requires to harvest the berries as late as possible, wait until the berry is about to die
+                if ((this.__internal__currentStrategy.harvestStrategy === this.__internal__harvestTimingType.RightBeforeWithering)
+                    && (this.__internal__getTimeUntilStage(plot, PlotStage.Berry) > 15))
+                {
+                    continue;
+                }
+                // If we are trying to unlock a new berry, dont harvest other berries so they can mutate
+                if ((this.__internal__currentStrategy.berryToUnlock !== undefined)
+                    && (plot.berryData.name !== this.__internal__currentStrategy.berryToUnlock))
```

Comment:
Why did you switch from `plot.berry` to `plot.berryData.name` here ?

The berry name is not te same as the berry type

Reply:
Just copied the wrong thing. Commiting it now.